### PR TITLE
Remind about AutoNonVariableTypeMode in error message.

### PR DESCRIPTION
**Pull request by ailzhang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51655 Remind about AutoNonVariableTypeMode in error message.**

Differential Revision: [D26228508](https://our.internmc.facebook.com/intern/diff/D26228508)

### CODE REVIEW

**Comment by ailzhang** [comment]
Improve error message to give more hints about skipping autograd (which is a common reason users hit this error).

**Review by xta0 — APPROVED** [review]
LGTM. Thanks for adding the message!

File: aten/src/ATen/core/boxing/KernelFunction.cpp [hunk 1]
```diff
@@ -23,6 +23,8 @@ void ambiguous_autogradother_kernel(OperatorKernel*, const OperatorHandle& op, S
   TORCH_INTERNAL_ASSERT(0,
     op.operator_name(), " has kernels registered to both Math and a backend mapped to AutogradOther. "
     "This makes the backend kernel unreachable (see Note [Ambiguity in AutogradOther kernel]). "
+    "If you only want to run inference instead of training, add `at::AutoNonVariableTypeMode guard(true);` "
+    "before model.forward().\n"
```
Comment (bhosmer) [inline aten/src/ATen/core/boxing/KernelFunction.cpp:27]: I'd actually move this new info to the end of the text, so the "If it's intended..." part relates more clearly to the AutogradOther kernel mentioned above.

Comment (bhosmer) [reply aten/src/ATen/core/boxing/KernelFunction.cpp:27]: Also, is it a problem that the advice is C++ specific?

**Review by bhosmer — APPROVED** [review]
Definitely nice to have the added info - couple things inline, gimme a shout if you wanna talk C++ vs Python messaging 😁

**Pull request closed without merging**